doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/logging.md

```diff
@@ -2,9 +2,149 @@
+patterns in the [Kubernetes documentation][KubDoc].
+
+### Sidecar containers
+We will provide an {ref}`example<kuberay-fluentbit>` of the sidecar strategy in this guide.
+You can process logs by configuring a log-processing sidecar
+for each Ray pod. Ray containers should be configured to share the `/tmp/ray`
+directory with the logging sidecar via a volume mount.
+
+You can configure the sidecar to do either of the following:
+* Stream Ray logs to the sidecar's stdout.
+* Export logs to an external service.
+
+### Daemonset
+Alternatively, it is possible to collect logs at the Kubernetes node level.
+To do this, one deploys a log-processing daemonset onto the Kubernetes cluster's
+nodes. With this strategy, it is key to mount
+the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
+
+(kuberay-fluentbit)=
+# Setting up logging sidecars with Fluent Bit.
+In this section, we give an example of how to set up log-emitting
+[Fluent Bit][FluentBit] sidecars for Ray pods.
+
+## Configure log processing
+The first step is to create a ConfigMap with configuration
+for FluentBit.
+
+Here is a minimal ConfigMap which tells a Fluent Bit sidecar to
+* Tail Ray logs.
+* Output the logs to the container's stdout.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: fluentbit-config
+data:
+  fluent-bit.conf: |
+    [INPUT]
+        Name tail
+        Path /tmp/ray/session_latest/logs/*
+        Tag ray
+        Path_Key true
+        Refresh_Interval 5
+    [OUTPUT]
+        Name stdout
+        Match *
+```
+In addition to streaming logs to stdout, you can export logs to any
+[storage backend][FluentBitStorage] supported by Fluent Bit.
+
+## Add logging sidecars to your RayCluster CR.
+
+### Add log and config volumes.
+For each pod template in our RayCluster CR, we
+need to add two volumes: One volume for Ray's logs
+and another volume to store Fluent Bit configuration from the ConfigMap
+applied above.
+```yaml
+volumes:
+- name: ray-logs
+  emptyDir: {}
+- name: fluentbit-config
+  configMap:
+    name: fluentbit-config
+```
+
+### Mount the Ray log directory
+Add the following volume mount to the Ray container's configuration.
+```yaml
+volumeMounts:
+- mountPath: /tmp/ray
+  name: ray-logs
+```
+
+### Add the Fluent Bit sidecar
+Finally, add the Fluent Bit sidecar container to each Ray pod config
+in your RayCluster CR.
+```yaml
+- name: fluentbit
+  image: fluent/fluent-bit:1.9.6
+  volumeMounts:
+  - mountPath: /tmp/ray
+    name: ray-logs
+  - mountPath: /fluent-bit/etc/fluent-bit.conf
+    name: fluentbit-config
+```
+Mounting the `ray-logs` volume gives the sidecar container access to Ray's logs.
+The `fluentbit-config` volume gives the sidecar access to logging configuration.
+
+(kuberay-logging-tldr)=
+## Putting everything together
+Now, we will see how to deploy the configuration described above.
+
+Deploy the KubeRay Operator if you haven't yet.
```

Comment:
It might be better to link to https://docs.ray.io/en/master/cluster/cluster_under_construction/ray-clusters-on-kubernetes/getting-started.html#deploying-the-kuberay-operator here -- this way in the future when we inevitably update this documentation we can be sure everything stays in sync :)

Reply:
"code reuse"

Reply:
Updated to link to getting started.